src/components/ArrayHelper.js

```diff
@@ -0,0 +1,57 @@
+import TypeErrorException from './BaseExceptions';
+import ObjectHelper from './ObjectHelper';
+
+class ArrayHelper {
+	static isEmpty( Arr ) {
+		return Arr.length === 0;
+	}
+
+	static copyArray( Arr, Depth = -1 ) {
+
+		if ( Array.isArray( Arr ) === false ) {
+			throw new TypeErrorException(
+				`Expected array got ${ typeof arguments[ 0 ] } at argument 0.`
+			);
+		}
+		return ObjectHelper.copyObj( Arr, Depth );
+	}
+
+	static mergeArray() {
```

Comment:
Could this possibly be removed since we have array and objects spreads now?

```
const foo = [1, 2]
const bar = [3, 4]
[...foo, ...bar]
// => [1, 2, 3, 4]

const omg = {foo: 'foo'}
const bbq = {bar: 'bar'}
{...omg, ...bbq}
// => {foo: 'foo', bar: 'bar'}
```

Reply:
Spreads do not perform a deep copy, but this method does that. So if we say a deep copy must performed explicit on one or more objects before we merge it, then I agree. It's more a question of policy, from my perspective.